src/Symfony/Component/Form/ChoiceList/ChoiceList.php

```diff
@@ -51,7 +51,7 @@ public static function lazy($formType, callable $choices, $vary = null): ChoiceL
      * @param ChoiceLoaderInterface                        $loader   A loader responsible for creating loading choices or grouped choices
      * @param mixed|null                                   $vary     Dynamic data used to compute a unique hash when caching the loader
      */
-    public static function loader($formType, ChoiceLoaderInterface $loader, $vary = null): ChoiceLoader
+    public static function loader(FormTypeInterface|FormTypeExtensionInterface $formType, ChoiceLoaderInterface $loader, $vary = null): ChoiceLoader
     {
         return new ChoiceLoader($formType, $loader, $vary);
```

Comment:
All constructors in this file (`ChoiceLoader`, `ChoiceValue`, `ChoiceFilter`, `ChoiceLabel`, `ChoiceFieldName`, `ChoiceAttr`, `ChoiceTranslationParameters`, `GroupBy` and `PreferredChoice`) can probably also get this union type